File: backend/MiningIssueTracker.Api/MiningIssueTracker.Api/Models/Issue.cs
namespace MiningIssueTracker.Api.Models
{
    public class Issue
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public IssuePriority Priority { get; set; } = IssuePriority.Medium;
        public IssueStatus Status { get; set; } = IssueStatus.Open;
        public int SiteId { get; set; }
        public int? EquipmentId { get; set; }
        public string? AssignedTo { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ResolvedAt { get; set; }
        public Site Site { get; set; } = null!;
        public Equipment? Equipment { get; set; }
    }
}

File: frontend/src/types/issue.ts
export const IssueStatus = {
  Open: 0,
  InProgress: 1,
  Resolved: 2,
} as const

export type IssueStatus = (typeof IssueStatus)[keyof typeof IssueStatus]

export const IssuePriority = {
  Low: 0,
  Medium: 1,
  High: 2,
} as const

export type IssuePriority = (typeof IssuePriority)[keyof typeof IssuePriority]

export interface Site {
  id: number
  name: string
  location?: string | null
}

export interface Equipment {
  id: number
  name: string
  type?: string | null
  siteId: number
}

export interface Issue {
  id: number
  title: string
  description: string
  category: string
  priority: IssuePriority
  status: IssueStatus
  siteId: number
  equipmentId?: number | null
  assignedTo?: string | null
  createdAt: string
  resolvedAt?: string | null
  site?: Site | null
  equipment?: Equipment | null
}

export interface CreateIssueRequest {
  title: string
  description: string
  category: string
  priority: IssuePriority
  siteId: number
  equipmentId?: number | null
  assignedTo?: string | null
}

export interface DashboardSummary {
  openIssues: number
  inProgressIssues: number
  highPriorityIssues: number
  averageResolutionHours: number
}

export interface IssueFilters {
  status?: IssueStatus
  priority?: IssuePriority
  siteId?: number
}
